numba/pythoncapi_compat.h

```diff
@@ -0,0 +1,1114 @@
+// Header file providing new C API functions to old Python versions.
+//
+// File distributed under the Zero Clause BSD (0BSD) license.
+// Copyright Contributors to the pythoncapi_compat project.
+//
+// Homepage:
+// https://github.com/python/pythoncapi_compat
+//
+// Latest version:
+// https://raw.githubusercontent.com/python/pythoncapi_compat/master/pythoncapi_compat.h
```

Comment:
I think it would be a good idea to include a reference with an explicit SHA in it such that we know a) which version this is and b) can track updates more easily as a result. What's in `HEAD` of the `pythoncapi_compat` repo has already drifted in comparison to this file.

Reply:
sync'ed in a651ac603c38eeaa70b556043369542f9338c2c2

Reply:
Checked as per https://github.com/numba/numba/pull/9682#pullrequestreview-2421717024.